test/helpers/helpers.go

```diff
@@ -124,6 +124,33 @@ func WaitForExpectedPhaseAndMessage(t *testing.T, cscName string, namespace stri
 	return err
 }
 
+// WaitForNotFound polls the cluster for a particular resource name and namespace
+// If the request fails because the runtime object is found it retries until the specified timeout
+// If the request returns a IsNotFound error the method will return true
+func WaitForNotFound(t *testing.T, result runtime.Object, namespace, name string) error {
```

Comment:
I went with `WaitForNotFound` because we only return true when the `errors.IsNotFound` call passes.

>I think this function is being introduced by a separate PR as well. I assume we are going to do some refactor to keep one?

Yep! We're pushing this PR before rebasing the others.